### ISSUE TITLE
passing trait reference fails inside (let ...)

### ISSUE THREAD

**Issue by psq** [blockstack/stacks-blockchain#1643]
**Describe the bug**
based on the test_passing_trait_reference_instances test, this works at execution
```
(define-trait trait-1 (
  (get-1 (uint) (response uint uint))
))

(define-public (wrapped-get-1 (value bool) (contract <trait-1>))
  (internal-get-1 contract)
)

(define-public (internal-get-1 (contract <trait-1>))
  (ok u1)
)
```

however, when the reference to the trait variable is done within `(let ...)` executing this code

```
(define-trait trait-1 (
  (get-1 (uint) (response uint uint))
))

(define-public (wrapped-get-1 (value bool) (contract <trait-1>))
  (let ((amount u0))  ;; added line, and its closing parenthesis
    (internal-get-1 contract)
  )
)

(define-public (internal-get-1 (contract <trait-1>))
  (ok u1)
)
```
fails with
```
Unchecked(UndefinedVariable("contract"))
       at NativeClarityBinProvider.execute (/Users/psq/src/perso/clarity-smart-contracts/node_modules/@blockstack/clarity/src/providers/clarityBin/index.ts:162:13)
       at processTicksAndRejections (internal/process/task_queues.js:86:5)
```

**Steps To Reproduce**
there is a more complex example in @friedger repo under https://github.com/friedger/clarity-smart-contracts/blob/feature/tax-office-trait/contracts/experiments/flip-coin-at-two.clar that can be run with
```
mocha test/unit/flip-coin-at-two.ts
```
But the contract snippet included can be used as is and calling `wrapped-get-1`

**Expected behavior**
it should not fail and find the the reference to the `contract` variable.

**Comment by lgalabru** [comment]
Thank you for this bug report @psq, I'll take a look.

**Comment by lgalabru** [comment]
Addressed with #1644
